src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -513,3 +556,47 @@ void GafferScene::EditScopeAlgo::removeParameterEdit( Gaffer::EditScope *scope,
 	// the edit instead.
 	edit->enabledPlug()->setValue( false );
 }
+
+const GraphComponent *GafferScene::EditScopeAlgo::parameterEditReadOnlyReason( const Gaffer::EditScope *scope, const ScenePlug::ScenePath &path, const std::string &attribute, const IECoreScene::ShaderNetwork::Parameter &parameter )
+{
+	auto *processor = acquireParameterProcessor( const_cast<EditScope *>( scope ), attribute, /* createIfNecessary = */ false );
+	if( !processor )
+	{
+		return MetadataAlgo::readOnlyReason( scope );
+	}
+
+	string pathString;
+	ScenePlug::pathToString( path, pathString );
+
+	auto *rows = processor->getChild<Spreadsheet::RowsPlug>( "edits" );
+	Spreadsheet::RowPlug *row = rows->row( pathString );
+	if( !row )
+	{
+		return MetadataAlgo::readOnlyReason( rows );
+	}
+
+	string tweakName = parameter.name.string();
+	if( parameter.shader.string().size() )
+	{
+		tweakName = parameter.shader.string() + "." + tweakName;
+	}
+
+	string columnName = boost::replace_all_copy( tweakName, ".", "_" );
+	auto *cell = row->cellsPlug()->getChild<Spreadsheet::CellPlug>( columnName );
+	if( !cell )
+	{
+		return MetadataAlgo::readOnlyReason( row->cellsPlug() );
+	}
+
+	const TweakPlug *tweak = cell->valuePlug<TweakPlug>();
+	if( const auto reason = MetadataAlgo::readOnlyReason( tweak->valuePlug<Plug>() ) )
+	{
+		return reason;
+	}
+	if( const auto reason = MetadataAlgo::readOnlyReason( tweak->enabledPlug() ) )
+	{
+		return reason;
+	}
```

Comment:
Should we be considering the children of `valuePlug()` here, and also `modePlug()`? If so, perhaps the recursive approach I outlined above is preferable?

Reply:
Improved [here](https://github.com/GafferHQ/gaffer/pull/3840/commits/de7a2934c5ede418ab0ac14ad9cafa7c9136fa98)